check_pgactivity

```diff
@@ -5600,6 +5604,186 @@ sub check_oldest_idlexact {
+                ELSE 'active_xact'
+              END
+          END AS kind,
+          datname,
+          coalesce(backend_xmin, backend_xid) AS xmin
+          FROM pg_stat_activity
+          -- exclude ourselves, as a blocked xmin in another database would be
+          -- exposed in the database we're connecting too, which may otherwise
+          -- not have the same xmin
+          WHERE pid != pg_backend_pid()
+          UNION ALL (
+            -- 2PC
+            SELECT '2pc' AS kind,
+            database AS datname,
+            transaction AS xmin
+            FROM  pg_prepared_xacts
+          ) UNION ALL (
+            -- replication slots
+            SELECT 'repslot' AS kind,
+            database AS datname,
+            xmin AS xmin
+            FROM  pg_replication_slots
+          ) UNION ALL (
+            -- walsenders
+            SELECT 'walsender' AS kind,
+            NULL AS datname,
+            backend_xmin AS xmin
+            FROM pg_stat_replication
+          )
+        ),
+        ordered AS (
+          SELECT f.kind, d.datname,
+          -- xid type doesn't have range operators as the value will wraparound.
+          -- Instead, rely on age() function and row_number() window function
+          -- to get the oldest xid found.
+          row_number() OVER (
+            PARTITION BY f.kind, d.datname
+            ORDER BY age(xmin) DESC NULLS LAST
+          ) rownum, age(xmin) AS age, xmin
+          FROM raw
+          RIGHT JOIN (VALUES
+            ( 'query'       ),
+            ( 'idle_xact'   ),
+            ( 'active_xact' ),
+            ( '2pc'         ),
+            ( 'repslot'     ),
+            ( 'walsender'   )
+          ) f(kind) ON raw.kind = f.kind
+          -- we use this JOIN condition to make sure that we'll always have a
+          -- full record for all (connectable) databases
+          JOIN pg_database d ON d.datname = coalesce(raw.datname, d.datname)
+          WHERE d.datallowconn
+        )
+        SELECT kind, datname, age, xmin
+        FROM ordered
+        WHERE rownum = 1
+        }
+    );
+
+    # warning and critical must be raw.
+    pod2usage(
+        -message => "FATAL: critical and warning thresholds only accept raw number of transactions.",
+        -exitval => 127
+    ) unless $args{'warning'}  =~ m/^([0-9.]+)$/
+        and  $args{'critical'} =~ m/^([0-9.]+)$/;
+
+    $c_limit = $args{'critical'};
+    $w_limit = $args{'warning'};
+
+    @hosts = @{ parse_hosts %args };
+
+    pod2usage(
+        -message => 'FATAL: you must give only one host with service "oldest_xmin".',
+        -exitval => 127
+    ) if @hosts != 1;
+
+    is_compat $hosts[0], 'oldest_xmin', $PG_VERSION_84 or exit 1;
+
+    @rs = @{ query_ver( $hosts[0], %queries ) };
+
+    REC_LOOP: foreach my $r (@rs) {
+        map { $_ = 'NaN' if $_ eq ''} @{$r}[2..3];
+        push @perfdata => (
+            ["$r->[0]_$r->[1]_age", $r->[2]],
+            ["$r->[0]_$r->[1]_xmin", $r->[3]]
```

Comment:
I still think reporting 6 kinds * 2 metrics * num_db  is very a lot of metrics, but anyway. I can give up on this as far as we can move on this PR. We'll be able to discuss and find solutions later if it is a real pb raising from the field.

However, using `"$r->[1]_$r->[0]_age` (`$db_$kind_age|xmin`) reads better to me. Eg. `dbname_walsender_age` instead of `walsender_dbname_age`. "age" is related to "kind". Moreover, it is easier to sort by dbname when checking metrics by hands.

Reply:
The only way to reduce then number of metrics is to report physical slots only once, as this is the only global metric.  That probably won't save much, but I won't object to that change.  Special care will have to be taken for the documentation to help users determining what is the xmin horizon on a specific database, as this won't be as obvious as before.

Reply:
As discussed, I'm ok with this.